src/backend/catalog/pg_constraint.c

```diff
@@ -836,6 +836,59 @@ get_constraint_relation_oids(Oid constraint_oid, Oid *conrelid, Oid *confrelid)
 	ReleaseSysCache(tup);
 }
 
+/*
+ * get_constraint_relation_columns
+ *		Find the columns of the relations to which a constraint refers.
+ *		Returns the list of found columns.
+ */
+List *
+get_constraint_relation_columns(Oid constraint_oid)
+{
+	List		*conkeys_list = NIL;
+	HeapTuple 	tp;
+	Relation	pg_constraint = heap_open(ConstraintRelationId, AccessShareLock);
+
+	tp = SearchSysCache1(CONSTROID, ObjectIdGetDatum(constraint_oid));
+	if (HeapTupleIsValid(tp))
+	{
+		bool 	is_null;
+		int16	*conkeys;
+		int		num_conkeys;
+		int 	i = 0;
+		Datum adatum = heap_getattr(tp, Anum_pg_constraint_conkey,
+							  		RelationGetDescr(pg_constraint), &is_null);
+
+		if (!is_null)
+		{
+			ArrayType  *arr = DatumGetArrayTypeP(adatum);
+
+			num_conkeys = ARR_DIMS(arr)[0];
```

Comment:
```suggestion
			int num_conkeys = ARR_DIMS(arr)[0];
```

Reply:
In the pg_constraint.c file usually all variables are declared at the beginning of the block, so it will be a difference comparing with nearby code style (which I think we try to avoid).

Reply:
This proposal does not seem to violate this agreement.